library/src/test/java/com/chuckerteam/chucker/api/ChuckerInterceptorTest.kt

```diff
@@ -562,6 +562,38 @@ internal class ChuckerInterceptorTest {
         )
     }
 
+    @ParameterizedTest
+    @EnumSource(value = ClientFactory::class)
+    fun `headers size are computed before being redacted`(factory: ClientFactory) {
+        val chuckerInterceptor = ChuckerInterceptorDelegate(
+            maxContentLength = SEGMENT_SIZE,
+            headersToRedact = setOf("Header-To-Redact"),
+            cacheDirectoryProvider = { tempDir },
+        )
+        val client = factory.create(chuckerInterceptor)
+
+        val request = Request.Builder().url(serverUrl).header("Header-To-Redact", "Hello").build()
+        val call = client.newCall(request)
+
+        server.enqueue(
+            MockResponse().addHeader("Header-To-Redact", "Goodbye").setResponseCode(HTTP_NO_CONTENT)
+        )
+        val response = call.execute()
+
+        val (originalRequestHeadersSize, originalResponseHeadersSize) = when (factory) {
+            ClientFactory.APPLICATION -> {
+                Pair(response.request.headers.byteCount(), response.headers.byteCount())
+            }
+            ClientFactory.NETWORK -> {
+                Pair(server.takeRequest().headers.byteCount(), response.networkResponse!!.headers.byteCount())
+            }
+        }
```

Comment:
I do insist that you use the suggested snippet for this. However, using `when` instead of `if/else` is probably better. This change introduces unnecessary duplication by extracting requests, headers and taking data from different places, blurring the whole picture.

Reply:
I changed a little but the problem of taking data from different sources is mantained... both request and response are changed depending of the `ClientFactory`...

Reply:
Code below passes the test. What is the problem?

```kotlin
val chuckerInterceptor = ChuckerInterceptorDelegate(
    maxContentLength = SEGMENT_SIZE,
    headersToRedact = setOf("Header-To-Redact"),
    cacheDirectoryProvider = { tempDir },
)
val client = factory.create(chuckerInterceptor)

val request = Request.Builder().url(serverUrl).header("Header-To-Redact", "Hello").build()
val call = client.newCall(request)

server.enqueue(
    MockResponse().addHeader("Header-To-Redact", "Goodbye").setResponseCode(HTTP_NO_CONTENT)
)
val response = call.execute()

val transaction = chuckerInterceptor.expectTransaction()
val expectedResponse = when (factory) {
    ClientFactory.APPLICATION -> response
    ClientFactory.NETWORK -> response.networkResponse!!
}
assertThat(transaction.requestHeadersSize).isEqualTo(expectedResponse.request.headers.byteCount())
assertThat(transaction.responseHeadersSize).isEqualTo(expectedResponse.headers.byteCount())
```

Reply:
Oh, now it worked! I probably did something wrong when I first tried it. Thanks!